lib/clib/CCorresLemmas.thy

```diff
@@ -990,6 +990,30 @@ lemma ccorres_cond2':
   apply clarsimp
   done
 
+text \<open>
+  A stronger version of @{thm ccorres_cond_both}, which allows the user to specify a condition on
+  the C state when showing equality of the guards of the conditionals, and which will provide the
+  relevant alternative on the truth of the guard of the @{term If} statement as an assumption in the
+  @{term ccorres_underlying} assumptions.\<close>
```

Comment:
Bit wordy, this one.
suggestion, something like: "
A stronger version of @{thm ccorres_cond_both}, which allows specifying a C state predicate
when showing equality of the conditionals' guards, and which supplies the relevant @{term If} guard truth to the @{term ccorres_underlying} assumptions.›"